torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -264,6 +267,81 @@ class SchedulerTopologyChecker {
 };
 } // namespace
 
+SchedulerRuntimeInfo::SchedulerRuntimeInfo(
+    Fusion* complete_fusion,
+    const at::ArrayRef<IValue>& inputs,
+    bool create_expr_evaluator)
+    : complete_fusion_(complete_fusion) {
+  collectAlignmentInfo(inputs);
+  if (create_expr_evaluator) {
+    initializeExpressionEvaluator(inputs);
+  }
+}
+
+SchedulerRuntimeInfo::SchedulerRuntimeInfo(
+    const SchedulerRuntimeInfo& copy_from)
+    : complete_fusion_(copy_from.complete_fusion_),
+      alignment_map_(copy_from.alignment_map_),
+      common_alignment_size_(copy_from.common_alignment_size_) {
+  expression_evaluator_ =
+      std::make_unique<ExpressionEvaluator>(complete_fusion_);
+}
+
+size_t SchedulerRuntimeInfo::getAlignmentSize(TensorView* tv) {
+  auto alignment_entry = alignment_map_.find(tv);
+  if (alignment_entry == alignment_map_.end()) {
+    return max_alignment_size_in_byte;
+  } else {
+    return alignment_entry->second;
+  }
+}
+
+void SchedulerRuntimeInfo::initializeExpressionEvaluator(
+    const at::ArrayRef<IValue>& inputs) {
+  expression_evaluator_ =
+      std::make_unique<ExpressionEvaluator>(complete_fusion_);
+  *expression_evaluator_ =
+      executor_utils::bindFusionInputs(inputs, complete_fusion_);
+}
+
+size_t SchedulerRuntimeInfo::collectAlignmentSize(
+    const at::Tensor& tensor) const {
+  // Essentially counting number of trailing 0's in the binary address
+  //  of the data ptr address
+
+  size_t alignment_size = 1;
+  size_t address = reinterpret_cast<size_t>(tensor.data_ptr());
+
+  const std::vector<std::pair<size_t, size_t>> binary_counting_table{
```

Comment:
at fun as bitwise math is, I'm having trouble understanding what you're going for here

Reply:
That was a bitwise math way of counting trailing zeros in the pointer address. I replaced it with a single for loop version. I guess since we are only counting up to 128b (7 bits), the benefit from bitwise math wasn't worth the confusion.